Library/Homebrew/cmd/test-bot.rb

```diff
@@ -480,6 +480,7 @@ def formula formula_name
       test "brew", "install", "--build-bottle", *changed_dependences unless changed_dependences.empty?
       formula_fetch_options = []
       formula_fetch_options << "--build-bottle" unless ARGV.include? "--no-bottle"
+      formula_fetch_options << "--build-from-source" unless ARGV.include? "--build-bottle"
```

Comment:
This is intended to catch the situations where people use the test-bot externally or locally and don't pass `--build-bottle`. The situation is likely to never pop up for us, but seemed a good idea to protect the eventuality and stop the situation where someone not passing `--build-bottle` gets the `modified formula` message chucked at them when they already know said formula is modified.